knative-operator/pkg/controller/knativekafka/knativekafka_controller.go

```diff
@@ -57,7 +60,59 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	// NOT IMPLEMENTED YET
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForOwner{
+		OwnerType:    &operatorv1alpha1.KnativeKafka{},
+		IsController: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	// common function to enqueue reconcile requests for resources
+	enqueueRequests := handler.ToRequestsFunc(func(obj handler.MapObject) []reconcile.Request {
+		annotations := obj.Meta.GetAnnotations()
+		ownerNamespace := annotations[common.KafkaOwnerNamespace]
+		ownerName := annotations[common.KafkaOwnerName]
+		if ownerNamespace != "" && ownerName != "" {
+			return []reconcile.Request{{
+				NamespacedName: types.NamespacedName{Namespace: ownerNamespace, Name: ownerName},
+			}}
+		}
+		return nil
+	})
+
+	gvkToResource := make(map[schema.GroupVersionKind]runtime.Object)
+
+	// Watch for Knative KafkaChannel resources.
+	kafkaChannelManifest, err := rawKafkaChannelManifest(mgr.GetClient())
```

Comment:
We're reading these a few times in the process of the reconciler. Should we maybe create the manifests just once at reconciler initialization and then always use the instance on the reconciler itself to avoid all the duplicated error handling. Plus it'd be a lot quicker.

Reply:
@markusthoemmes  please have a look at this comment: https://github.com/openshift-knative/serverless-operator/pull/496/commits/41ff1ae8b369f62c883775c8cb51a0d9be135f2c

I did it, but kindaa turned ugly

Reply:
How so? I like it!

Reply:
haha ok, everybody has a different taste :D